src/Idris/Elab/Interface.idr

```diff
@@ -128,8 +124,8 @@ getMethToplevel {vars} env vis iname cname constraints allmeths params
          -- Make the constraint application explicit for any method names
          -- which appear in other method types
          let ty_constr =
-             bindPs params $ substNames vars (map applyCon allmeths) ty
-         ty_imp <- bindTypeNames [] vars (bindIFace fc ity ty_constr)
+             substNames vars (map applyCon allmeths) ty
+         ty_imp <- bindTypeNames [] vars (bindPs params $ bindIFace fc ity ty_constr)
```

Comment:
BTW, is `bindIface` used anywhere else? Would it be easier to inline it here, now that it's no longer a pattern-matching recursive definition?

Reply:
`bindIface` isn't used anywhere else. I just left it in there because I thought it's better to change the code as little as possible, and because inlining the definition makes the code look a bit messy. On the other hand, it looks a bit odd to have such a trivial function, so I'm happy to inline it if you want.